pkg/protocol/commitment_verifier.go

```diff
@@ -7,21 +7,31 @@ import (
 	"github.com/iotaledger/hive.go/kvstore/mapdb"
 	"github.com/iotaledger/iota-core/pkg/model"
 	"github.com/iotaledger/iota-core/pkg/protocol/engine"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/accounts"
 	iotago "github.com/iotaledger/iota.go/v4"
 	"github.com/iotaledger/iota.go/v4/merklehasher"
 )
 
 type CommitmentVerifier struct {
-	engine *engine.Engine
+	engine                  *engine.Engine
+	forkingPoint            *model.Commitment
+	cumulativeWeight        uint64
+	validatorAccountsAtFork map[iotago.AccountID]*accounts.AccountData
 }
 
-func NewCommitmentVerifier(mainEngine *engine.Engine) *CommitmentVerifier {
+func NewCommitmentVerifier(mainEngine *engine.Engine, forkingPoint *model.Commitment) *CommitmentVerifier {
+	committeeAtForkingPoint := mainEngine.SybilProtection.SeatManager().Committee(forkingPoint.Index()).Accounts().IDs()
+
 	return &CommitmentVerifier{
-		engine: mainEngine,
+		engine:                  mainEngine,
+		forkingPoint:            forkingPoint,
+		cumulativeWeight:        forkingPoint.CumulativeWeight(),
+		validatorAccountsAtFork: mainEngine.Ledger.PastAccounts(committeeAtForkingPoint, forkingPoint.Index()),
+		// TODO: what happens if the committee rotated after the fork?
 	}
 }
 
-func (c *CommitmentVerifier) verifyCommitment(prevCommitment, commitment *model.Commitment, attestations []*iotago.Attestation, merkleProof *merklehasher.Proof[iotago.Identifier]) (iotago.BlockIDs, error) {
+func (c *CommitmentVerifier) verifyCommitment(commitment *model.Commitment, attestations []*iotago.Attestation, merkleProof *merklehasher.Proof[iotago.Identifier]) (iotago.BlockIDs, uint64, error) {
```

Comment:
do you want to give the return parameters some names for readability? (but do not use those vars ;-) )